pkg/cloud/services/wait/wait.go

```diff
@@ -84,9 +87,14 @@ func WaitForWithRetryable(backoff wait.Backoff, condition wait.ConditionFunc, re
 		return nil
 	}
 
-	// error occurred, return the actual error
-	if actualErr != nil {
-		return actualErr
+	// error is non-retryable
+	if waitErr != wait.ErrWaitTimeout {
+		return waitErr
+	}
+
+	// we are getting retryable errors always, timeout occurred
+	if errToReturn != nil {
+		return errToReturn
 	}
 
 	// no error occurred inside condition(), the error was timeout error
```

Comment:
I was working on a similar PR :joy:

I had removed the initial check for `waitErr == nil` and I had slightly different comments here, which I think are a bit clearer:
```go
	// If the waitError is not a timeout error (nil or a non-retryable error), return it
	if waitErr != wait.ErrWaitTimeout {
		return waitErr
	}

	// A retryable error occurred, return the actual error
	if actualErr != nil {
		return actualErr
	}

	// The error was timeout error
	return waitErr
```

Reply:
yes they're lot clearer! I updated the pr, ptal :D